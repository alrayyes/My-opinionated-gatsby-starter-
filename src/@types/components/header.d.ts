export interface HeaderInterface {
  siteTitle: string
}
