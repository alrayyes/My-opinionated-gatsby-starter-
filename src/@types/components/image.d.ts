import { FluidObject } from "gatsby-image"

export interface ImageDataInterface {
  placeholderImage: {
    childImageSharp: {
      fluid: FluidObject
    }
  }
}
