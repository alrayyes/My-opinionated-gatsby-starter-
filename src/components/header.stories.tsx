import { storiesOf } from "@storybook/react"
import React from "react"
import Header from "./header"

storiesOf("Header component", module).add(`default`, () => (
  <Header siteTitle={"My Opinionated Gatsby Default Starter"} />
))
