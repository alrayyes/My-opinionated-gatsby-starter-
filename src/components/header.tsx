import { Link } from "gatsby"
import React from "react"
import { HeaderInterface } from "../@types/components/header"
import styled from "@emotion/styled"

const Header: React.FC<HeaderInterface> = ({ siteTitle }) => {
  const Header = styled.header`
    background: rebeccapurple;
    margin-bottom: 1.45rem;
  `

  const Container = styled.div`
    margin: 0 auto;
    max-width: 960;
    padding: 1.45rem 1.0875rem;
  `

  const H1 = styled.h1`
    margin: 0;
  `

  const StyledLink = styled(Link)`
    color: white;
    text-decoration: none;
  `

  return (
    <Header>
      <Container>
        <H1>
          <StyledLink to="/">{siteTitle}</StyledLink>
        </H1>
      </Container>
    </Header>
  )
}

export default Header
