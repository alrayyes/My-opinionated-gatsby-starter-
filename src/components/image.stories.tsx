import { storiesOf } from "@storybook/react"
import React from "react"
import Image from "./image"

storiesOf("Image component", module).add(`default`, () => <Image />)
