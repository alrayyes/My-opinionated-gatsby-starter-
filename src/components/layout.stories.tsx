import { storiesOf } from "@storybook/react"
import React from "react"
import Layout from "./layout"

storiesOf("Layout component", module).add(`default`, () => (
  <Layout>content</Layout>
))
