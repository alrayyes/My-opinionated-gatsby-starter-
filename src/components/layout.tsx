/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React from "react"
import { useStaticQuery, graphql } from "gatsby"

import Header from "./header"
import "../utils/layout.css"
import {
  LayoutDataInterface,
  LayoutInterface,
} from "../@types/components/layout"

const Layout: React.FC<LayoutInterface> = ({ children }) => {
  const data: LayoutDataInterface = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return (
    <>
      <Header siteTitle={data.site.siteMetadata.title} />
      <div className={"layoutContainer"}>
        <main>{children}</main>
        <footer>
          Â© {new Date().getFullYear()}, Built with{" "}
          <a href="https://www.gatsbyjs.org">Gatsby</a>
        </footer>
      </div>
    </>
  )
}

export default Layout
