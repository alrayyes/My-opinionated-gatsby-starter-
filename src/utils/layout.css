@tailwind base;

div.layoutContainer {
  @apply m-0 m-auto max-w-screen-md pt-6 pr-4;
}

div.headerContainer {
  @apply m-0 m-auto max-w-screen-md pt-6 pr-4 pb-6;
}

a.headerLink {
  @apply text-white no-underline text-4xl font-black;
}

a {
  @apply font-bold text-blue-600 underline;
}

h1 {
  @apply text-4xl font-bold;
}

p {
  @apply mt-6 text-xl;
}

footer {
  @apply mt-8;
}
