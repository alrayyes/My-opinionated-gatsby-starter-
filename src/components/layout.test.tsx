import renderer from "react-test-renderer"
import React from "react"
import * as Gatsby from "gatsby"
import Layout from "./layout"
import { LayoutDataInterface } from "../@types/components/layout"

const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")

describe("Layout", () => {
  beforeEach(() => {
    const data: LayoutDataInterface = {
      site: {
        siteMetadata: {
          title: "Gatsby Default Starter",
        },
      },
    }

    useStaticQuery.mockImplementation(() => data)
  })

  it("renders correctly", () => {
    const tree = renderer.create(<Layout>children</Layout>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
