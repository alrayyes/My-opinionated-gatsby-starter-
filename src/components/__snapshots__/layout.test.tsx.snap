// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1pcnel4-Header{background:rebeccapurple;margin-bottom:1.45rem;}",
      }
    }
    data-emotion-css="1pcnel4-Header"
  />,
  <header
    className="css-1pcnel4-Header e1qr1m9j0"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-pf9rd5-Container{margin:0 auto;max-width:960;padding:1.45rem 1.0875rem;}",
        }
      }
      data-emotion-css="pf9rd5-Container"
    />
    <div
      className="css-pf9rd5-Container e1qr1m9j1"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1ikcnub-H1{margin:0;}",
          }
        }
        data-emotion-css="1ikcnub-H1"
      />
      <h1
        className="css-1ikcnub-H1 e1qr1m9j2"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-rkbe0q-StyledLink{color:white;-webkit-text-decoration:none;text-decoration:none;}",
            }
          }
          data-emotion-css="rkbe0q-StyledLink"
        />
        <a
          className="css-rkbe0q-StyledLink e1qr1m9j3"
          href="/"
        >
          Gatsby Default Starter
        </a>
      </h1>
    </div>
  </header>,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-15zhpu-Container{margin:0 auto;max-width:960;padding:0 1.0875rem 1.45rem;}",
      }
    }
    data-emotion-css="15zhpu-Container"
  />,
  <div
    className="css-15zhpu-Container e9jw82o0"
  >
    <main>
      children
    </main>
    <footer>
      Â© 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
